### PR TITLE
init test cases for rancher p

### PR DESCRIPTION
**Pull request by slickwarren**
## Issue: 

https://github.com/rancher/qa-tasks/issues/544
 
## Problem

we need some automation for checking rancher p
 
## Solution

This PR includes checking /rancherversion and ui-brand, ensuring that it matches what we expect. This can also be used ( and may need to be checked ) for the community version. 
 
## Testing

local run is passing, will post jenkins run separately

### CODE REVIEW

File: tests/v2/validation/prime/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+In your config file, set the following:
+```json
+"rancher": { 
+  "host": "rancher_server_address",
+  "adminToken": "rancher_admin_token"
+  ...,
+}
+"prime": {
+  "rancherVersion": "<version_or_commit_of_rancher",
```
Comment (markusewalker) [inline tests/v2/validation/prime/README.md:14]: nit..missing a `>` bracket at the end of the `rancherVersion`. Also, if you enter the version, you have to put a leading `v`, correct?

File: tests/v2/validation/prime/prime_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func (t *PrimeVersionTestSuite) TearDownSuite() {
+	t.session.Cleanup()
+}
+
+func (t *PrimeVersionTestSuite) SetupSuite() {
+	testSession := session.NewSession(t.T())
+	t.session = testSession
+	primeConfig := new(Config)
```
Comment (markusewalker) [inline tests/v2/validation/prime/prime_test.go:34]: nit - maybe we can break this up so it's not as cluttered? So a space after line 33 and a space after line 35?

File: tests/v2/validation/prime/prime_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+}
+
+func (t *PrimeVersionTestSuite) TestPrimeVersion() {
+	var http_url = "https://" + t.client.RancherConfig.Host + "/rancherversion"
+	req, err := http.Get(http_url)
+	if err != nil {
+		t.T().Log(err)
+	}
+	byte_body, err := ioutil.ReadAll(req.Body)
```
Comment (markusewalker) [inline tests/v2/validation/prime/prime_test.go:51]: same nit as above, can we put a space after line 50?

**Comment by markusewalker** [comment]
Overall, lgtm, just left some nits.

File: tests/v2/validation/prime/prime_test.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	t.rancherVersion = primeConfig.RancherVersion
+
+	client, err := rancher.NewClient("", t.session)
+	assert.NoError(t.T(), err)
+	t.client = client
+
+}
+
+func (t *PrimeVersionTestSuite) TestPrimeVersion() {
+	var http_url = "https://" + t.client.RancherConfig.Host + "/rancherversion"
```
Comment (igomez06) [inline tests/v2/validation/prime/prime_test.go:46]: Caleb I'm suggestion is, I think this whole block where you're checking the rancher version, and rancher prime bool should be in an extension called GetRancherVersion or something like that. Not in the test itself

File: tests/v2/validation/prime/README.md (hunk 1 above)
Comment (slickwarren) [reply tests/v2/validation/prime/README.md:14]: correct, however you can also enter a commit version so I didn't want to confuse the user if they plan to enter the commit version (which doesn't have a `v` in front).

**Review by markusewalker — COMMENTED** [review]
@slickwarren did we have a Jenkins run with your latest changes? The comments I had were resolved, everything else lgtm.

File: tests/framework/extensions/rancherversion/rancherversion.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package rancherversion
+
+import (
+	"io/ioutil"
+	"net/http"
+)
+
```
Comment (igomez06) [inline tests/framework/extensions/rancherversion/rancherversion.go:7]: Please put a comment here describing the function, so we can have a google doc on it.

File: tests/v2/validation/prime/prime_test.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	assert.NoError(t.T(), err)
+	t.client = client
+
+}
+
+func (t *PrimeVersionTestSuite) TestPrimeVersion() {
+	byte_object, err := rancherversion.RequestRancherversion(t.client.RancherConfig.Host)
+	assert.Nil(t.T(), err)
+
+	var jsonObject map[string]interface{}
```
Comment (igomez06) [inline tests/v2/validation/prime/prime_test.go:49]: The unmarshaling from 49-51 should be in the extension you wrote. I don't think logging the json brings any value, maybe the version so when it fails or passes we know what version it is.

File: tests/framework/extensions/rancherversion/rancherversion.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+/* Requests the rancher version from the rancher server, parses the returned
+ * json and returns a Struct object, or an error.
+ */
+func RequestRancherVersion(rancher_url string) (Config, error) {
```
Comment (igomez06) [inline tests/framework/extensions/rancherversion/rancherversion.go:13]: Return the pointer Caleb, so you don't have dereference at the Botton on line 30

File: tests/framework/extensions/rancherversion/rancherversion.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	req, err := http.Get(http_url)
+	if err != nil {
+		return *config_object, err
+	}
+	byte_object, err := ioutil.ReadAll(req.Body)
+	if err != nil || byte_object == nil {
+		return *config_object, err
+	}
+	var jsonObject map[string]interface{}
+	err = json.Unmarshal(byte_object, &jsonObject)
```
Comment (igomez06) [inline tests/framework/extensions/rancherversion/rancherversion.go:25]: Need an error check?

**Review by markusewalker — DISMISSED** [review]
LGTM

**Review by igomez06 — COMMENTED** [review]
LGTM

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by markusewalker — APPROVED** [review]
LGTM

**Pull request merged**